resources/assets/js/components/suggest-topic.vue

```diff
@@ -21,26 +21,31 @@
 </template>
 
 <script>
+    import Topics from './../topics.js';
+
     export default {
         data: function () {
             return {
-                title: '',
-                description: ''
+                topic: {
+                    title: '',
+                    description: '',
+                }
             };
         },
-        props: ['topics'],
         methods: {
             suggestTopic: function () {
                 var vm = this;
 
-                this.$http.post('topics', { title: this.title, description: this.description }, function (data) {
-                    vm.title = '';
-                    vm.description = '';
-
-                    vm.$dispatch('topics.created', data);
-
-                    vm.$route.router.go('/');
-                });
+                Topics.add(vm.topic).then(
+                        topic => {
```

Comment:
I'm pretty new to promises, so I'm not the right one to say. That having been said.. lol.. I think the resolving callbacks have access to the topics list already, right? But this is the only place they could possibly get access to the response data, which I presume means it's the only way they could technically do something in response to the one specific topic that was added. In that case, it seems like returning response here is the most future friendly *and* the most consistent? But again, I'm new at this.